taxcalc/tbi/tbi_utils.py

```diff
@@ -89,21 +93,53 @@ def dropq_calculate(year_n, start_year,
     growdiff_baseline.apply_to(growfactors_post)
     growdiff_response.apply_to(growfactors_post)
 
-    # create pre-reform Calculator instance using PUF input data & weights
-    recs1 = Records(data=copy.deepcopy(taxrec_df),
-                    gfactors=growfactors_pre)
+    # create sample pd.DataFrame from specified input file and sampling scheme
+    stime = time.time()
+    tbi_path = os.path.abspath(os.path.dirname(__file__))
+    if use_puf_not_cps:
+        # first try TaxBrain deployment path
+        input_path = 'puf.csv.gz'
+        if not os.path.isfile(input_path):
+            # otherwise try local Tax-Calculator deployment path
+            input_path = os.path.join(tbi_path, '..', '..', 'puf.csv')
```

Comment:
@martinholmer Currently, the PUF is sent to `taxcalc` as a pandas dataframe.  I'm not sure how to generally specify the PUF path relative to this file.  It may be easier for `use_puf_not_cps` to be a keyword argument where it is the PUF data frame if the user chooses the PUF file option and `None` or `False` otherwise.

Reply:
@hdoupe said:
> Currently, the PUF is sent to taxcalc as a pandas dataframe. I'm not sure how to generally specify the PUF path relative to this file.

The newest `tbi_utils.py` code tries to anticipate where the input files will be located.
The following code fragment from `PolicyBrain/webapp/apps/taxbrain/tasks.py` suggests to me that `puf.csv.gz` will be in the current working directory:
```
def get_tax_results_async(mods, inputs_pk):
    print "mods is ", mods
    user_mods = package_up_vars(mods)
    print "user_mods is ", user_mods
    print "begin work"
    cur_path = os.path.abspath(os.path.dirname(__file__))
    tax_dta = pd.read_csv("puf.csv.gz", compression='gzip')
    mY_dec, mX_dec, df_dec, mY_bin, mX_bin, df_bin, fiscal_tots = dropq.run_models(tax_dta,
        num_years=NUM_BUDGET_YEARS, user_mods={START_YEAR:user_mods})
```
And that is where the `tbi_utils.py` code looks for it.
Does this make sense?